tracer/src/Datadog.Trace/AspNet/SharedItems.cs

```diff
@@ -54,6 +54,21 @@ internal static Scope TryPopScope(HttpContext context, string key)
 
             return default(Scope);
         }
+
+        internal static Scope TryPeakScope(HttpContext context, string key)
+        {
+            var item = context?.Items[key];
```

Comment:
nit-We could factorize TryPeek and TryPop

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2495/commits/6785f56e8af6d600e01818116fa36b76c7e0cdb6

Reply:
I thought more of something like this
  
        private static readonly Func<Stack<Scope>, Scope> Pop = stack => stack.Pop();
        private static readonly Func<Stack<Scope>, Scope> Peek = stack => stack.Peek();
        internal static Scope TryPopScope(HttpContext context, string key) => ExtractScope(context, key, Pop);

        internal static Scope TryPeekScope(HttpContext context, string key) => ExtractScope(context, key, Peek);

        private static Scope ExtractScope(HttpContext context, string key, Func<Stack<Scope>, Scope> getter)
        {
            var item = context?.Items[key];
            if (item is Scope storedScope)
            {
                return storedScope;
            }
            else if (item is Stack<Scope> stack && stack.Count > 0)
            {
                return getter(stack);
            }

            return default(Scope);
        }`

But yours is two allocations less, so I assume better ? (TBH i'm not clear on how careful we should be w/ small static allocations)

Reply:
makes sense! fixed in [19542d8](https://github.com/DataDog/dd-trace-dotnet/pull/2495/commits/19542d8fac49caca13d79359a0712f996ed11add)